### PR TITLE
Move 5 static functions out of DartTypeUtilities

### PR DESCRIPTION
**Pull request by srawlins**
# Description

Move 5 static functions out of DartTypeUtilities. Helps linter comply with [AVOID defining a class that contains only static members.](https://dart.dev/guides/language/effective-dart/design#avoid-defining-a-class-that-contains-only-static-members) and makes the code much more readable.

* `canonicalElementsAreEqual` can be a top-level function
* `canonicalElementsFromIdentifiersAreEqual` can be a top-level function
* `getCanonicalElement` can be an extension on `Element` and be shortened to `canonicalElement` 🎉 
* `getCanonicalElementFromIdentifier` can be an extension on `AstNode` and renamed to be much shorter: `canonicalElement` 🎉 
* `isNullLiteral` can be an extension on `Expression`

### CODE REVIEW

**Comment by pq** [comment]
From a style perspective, I'm totally into this (and there are other changes we could make too).  Having said that, this is kinda-sorta API insofar as Google3-internal lints may get broken by a move.  Maybe consider a quick codesearch to see?

A more polite approach might be to deprecate these methods with a threat to remove them real soon.  (In practice this may be nearly just as breaking though...  Let's discuss!)

**Comment by srawlins** [comment]
I'll replace with deprecated forwarders for anything used in google3.

**Comment by srawlins** [comment]
* `canonicalElementsAreEqual`, `canonicalElementsFromIdentifiersAreEqual`, `getCanonicalElement`: 0 uses
* `getCanonicalElementFromIdentifier`, `isNullLiteral`: 2 uses each

**Comment by srawlins** [comment]
Added deprecated replacements for the two with references in google3.

**Review by bwilkerson — APPROVED** [review]

File: lib/src/util/dart_type_utilities.dart [hunk 1]
```diff
@@ -595,6 +558,48 @@ class InterfaceTypeDefinition {
   }
 }
 
+extension ElementExtensions on Element {
```
Comment (pq) [inline lib/src/util/dart_type_utilities.dart:568]: Consider moving these into `ast.dart`?  Or maybe introducing a new `extensions.dart` like we have in analyzer?

**Review by pq — APPROVED** [review]

File: lib/src/util/dart_type_utilities.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/util/dart_type_utilities.dart]: Ah ha I hadn't seen that file. Done!

**Pull request merged**